include/aws/http/request_response.h

```diff
@@ -233,6 +233,12 @@ struct aws_http_make_request_options {
      * See `aws_http_on_stream_complete_fn`.
      */
     aws_http_on_stream_complete_fn *on_complete;
+
+    /**
+     * When using h2, request body data will be provided over time. The stream will only be polled for writing
+     * when data has been supplied via `aws_h2_stream_write_data`
+     */
+    bool h2_use_manual_data_writes;
```

Comment:
Do we protect anywhere that this is valid (i.e. we actually got an H2 connection) if set to `true` after the connection is established?

Reply:
it's more than it's only read by H2 code, the same way we do HTTP/1.1 chunk extensions. Also, this cannot be changed on the stream, this is only for the request, it's effectively temporary (only for the `make_request` call). The request will copy it into the stream if the stream is H2.